### PR TITLE
Multiple Forging Nodes Test

### LINKED ISSUE #994: [Byzantine] Multiple forging nodes
This test should take a group of nodes that are initialized with keys that have been given an equal number of arbits to then start forging. The test should be ran for enough time so that each node has a chance to generate a block, and the assertion should decide that the distribution of block generation is relatively equal to the number of nodes.

### LINKED ISSUE #1129: Replace Blocking in HttpService
HttpService defines a `basicRoute` method which returns an akka-http route.  In the implementation, a Future is created by the API Service Handler(s).  akka-http allows for marshaling of Futures, meaning we do not need to run `Await.result(...)` and can instead return a `Future[HttpResponse]` (or similar).

To still accomplish a timeout mechanism, a pattern like this can be used:
```scala
import akka.pattern.after

val someLongFuture: Future[HttpResponse] = ???

Future.firstCompletedOf(
    List(
        someLongFuture,
        akka.pattern.after(10.seconds)(Future.successful(SomeErrorHttpResponse())
    )
)
```

The use of `Await.result` should be avoided at all costs, with the exception of service startup or shutdown, or in test cases.  Blocking code running in an async execution context can significantly deteriorate performance.

Acceptance Criteria:
- HttpService's `basicRoute` method does not use a blocking `Await.result`.

```

### PR DESCRIPTION
**Pull request by SeanCheatham**
## Purpose
- An integration test that verifies basic networking, peer, and forging behavior between multiple Bifrost node
- Also verifies "fairness" of block forging distribution

## Approach
- Created a new test suite which:
    - Creates 3 nodes (with forging disabled) in a shared Docker network
    - "Assigns" a forging address to each node
    - Instructs each node to forge for 5 minutes
    - Verifies that each node forged a roughly equal number of blocks
- In addition, I fixed the "Await" in HttpService since I may have been hitting the thread contention issue when writing these tests

## Testing
- The test runs and "works", but the distribution of forged blocks is not necessarily equal
- Test run with 2 minutes of forging:
```
14:37:47.692| INFO  [pool-8-thread-8-ScalaTest-running-MultiNodeTest] co.topl.it.MultiNodeTest - myBlocks forging count=21 containerId=68b6fee85c34c107ea63926686ea42a6bab2bab87d137cc6f21c02f4d1c9939c
14:37:47.692| INFO  [pool-8-thread-8-ScalaTest-running-MultiNodeTest] co.topl.it.MultiNodeTest - myBlocks forging count=34 containerId=64a86943965c930ba222429a88bc236d4f21e34d6823874dcd86f3dfc310c3b1
14:37:47.692| INFO  [pool-8-thread-8-ScalaTest-running-MultiNodeTest] co.topl.it.MultiNodeTest - myBlocks forging count=15 containerId=a72213adc3f89208d34add8ed787b276e050fce4a86e06578e438efbf50b7ff8
14:37:47.712| INFO  [pool-8-thread-8-ScalaTest-running-MultiNodeTest] co.topl.it.MultiNodeTest - Final block generator counts for containerId=68b6fee85c34c107ea63926686ea42a6bab2bab87d137cc6f21c02f4d1c9939c:
{
  "AU9NT19nugfYTGv8w6s4ALTF1Df6LaiU1xgd4pVLsf5oTxyQJfAW" : 21,
  "AU9Qsd93BwEf8uUU95RZ8nAZApaU6KsgFhJyRqEugezfZsARt3k8" : 15,
  "AUA6qr3AveZpfB463BwipAVUzRtJv9G1EJMHSy7BrY5HKnpgC9s4" : 34,
  "AUAx8QEMPvX8wd3m2VaipEXvxqM5jUDEDa5rx2UbA2RMbqU77fae" : 1
}
14:37:47.712| INFO  [pool-8-thread-8-ScalaTest-running-MultiNodeTest] co.topl.it.MultiNodeTest - Final block generator counts for containerId=64a86943965c930ba222429a88bc236d4f21e34d6823874dcd86f3dfc310c3b1:
{
  "AU9NT19nugfYTGv8w6s4ALTF1Df6LaiU1xgd4pVLsf5oTxyQJfAW" : 21,
  "AU9Qsd93BwEf8uUU95RZ8nAZApaU6KsgFhJyRqEugezfZsARt3k8" : 15,
  "AUA6qr3AveZpfB463BwipAVUzRtJv9G1EJMHSy7BrY5HKnpgC9s4" : 34,
  "AUAx8QEMPvX8wd3m2VaipEXvxqM5jUDEDa5rx2UbA2RMbqU77fae" : 1
}
14:37:47.712| INFO  [pool-8-thread-8-ScalaTest-running-MultiNodeTest] co.topl.it.MultiNodeTest - Final block generator counts for containerId=a72213adc3f89208d34add8ed787b276e050fce4a86e06578e438efbf50b7ff8:
{
  "AU9NT19nugfYTGv8w6s4ALTF1Df6LaiU1xgd4pVLsf5oTxyQJfAW" : 21,
  "AU9Qsd93BwEf8uUU95RZ8nAZApaU6KsgFhJyRqEugezfZsARt3k8" : 15,
  "AUA6qr3AveZpfB463BwipAVUzRtJv9G1EJMHSy7BrY5HKnpgC9s4" : 34,
  "AUAx8QEMPvX8wd3m2VaipEXvxqM5jUDEDa5rx2UbA2RMbqU77fae" : 1
}
[info] MultiNodeTest:
[info] - Multiple nodes can forge blocks with roughly equal distribution
[info] ScalaTest
[info] Run completed in 2 minutes, 28 seconds.
[info] Total number of tests run: 2
[info] Suites: completed 2, aborted 0
[info] Tests: succeeded 2, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 2, Failed 0, Errors 0, Passed 2
[success] Total time: 169 s (02:49), completed Mar 30, 2021 2:37:49 PM
```
- Block distribution with 5 minutes of forging:
```
  "AUATfFNeg2pJFn1P2oxiZz2awdekavUgVxR3TM2w6Nom9gow1MBK" : 42,
  "AUAVfeSvg7MA1NhmFpNe9LzXhZSoWxVsqHrmDKz2PHj3Mc6tSDyr" : 43,
  "AUAx8QEMPvX8wd3m2VaipEXvxqM5jUDEDa5rx2UbA2RMbqU77fae" : 1,
  "AUB9WodZjFK5y3QSteNpkx3GHmL7buyGc7SjVrLxonjQaSpLqZuZ" : 44

```
- Block distribution with 10 minutes of forging:
```
  "AU9rG3vTBznCQ7d1Hve5AHwxj8ifPN2fsfRz1Degrq8aS8qx1FQg" : 81,
  "AUAnrJHa6nfs5AwGSXM7QQo67CZ4o9o2t3EwWFcu3tiY891mmMLn" : 86,
  "AUAx8QEMPvX8wd3m2VaipEXvxqM5jUDEDa5rx2UbA2RMbqU77fae" : 1,
  "AUB9XjET4QRDGk3NLUmqHxwxxVkt2FAiDSbE4QPwuSkJmChzRRBx" : 60
  ```
  
  ## Tickets
  - closes #1129
  - closes #994

### CODE REVIEW

File: src/main/scala/co/topl/settings/StartupOpts.scala [hunk 1]
```diff
@@ -47,7 +47,11 @@ final case class RuntimeOpts(
     val rpcApiSettings = appSettings.rpcApi.copy(
       apiKeyHash = apiKeyHash.fold[String](appSettings.rpcApi.apiKeyHash)(a => a)
     )
-    val privateTestnetSettings = appSettings.forging.privateTestnet.map(_.copy(genesisSeed = seed))
```
Comment (SeanCheatham) [inline src/main/scala/co/topl/settings/StartupOpts.scala:50]: I noticed that the `seed` optional startup option would _always_ override the genesisSeed in AppSettings, even if `None` was provided.  This made it impossible to set a seed via configuration.

File: it/src/test/scala/co/topl/it/util/DockerSupport.scala [hunk 2]
```diff
@@ -4,27 +4,48 @@ import akka.actor.ActorSystem
...
+  private def buildContainerConfig(name: String, environment: Map[String, String]): ContainerConfig = {
     val env =
       environment.toList.map { case (key, value) => s"$key=$value" }
 
     val cmd =
-      List("--apiKeyHash", NodeRpcApi.ApiKeyHashBase58, "-c", "/opt/docker/config/testConfig.conf", "-s", seed)
+      List("--apiKeyHash", NodeRpcApi.ApiKeyHashBase58, "-c", "/opt/docker/config/testConfig.conf")
+
+    val hostConfig =
+      HostConfig.builder().nanoCpus((1d * 1000000000).toLong).build()
```
Comment (SeanCheatham) [inline it/src/test/scala/co/topl/it/util/DockerSupport.scala:48]: This sets the number of CPU cores allocated to each node to `1`.  I did this to enforce some sort of CPU consistency between nodes.  We could potentially add another multi-node test in the future which creates a node with `0.1` CPU and a node with `2` CPU and verify that they're still awarded a fair distribution of blocks.

Comment (scasplte2) [reply it/src/test/scala/co/topl/it/util/DockerSupport.scala:48]: That is interesting that you can control this. Have you played with this at all and/or seen any difference in distribution?

File: src/main/scala/co/topl/settings/StartupOpts.scala (hunk 1 above)
Comment (scasplte2) [reply src/main/scala/co/topl/settings/StartupOpts.scala:50]: oh, good catch! I think @jingsting had been encountering this and I didn't understand what was causing it.

**Review by scasplte2 — APPROVED** [review]
👍 nicely done

**Review by b-vennes — APPROVED** [review]

**Pull request merged**